test/commands/deploy.test.ts

```diff
@@ -243,3 +219,145 @@ describe('pollPipeline handles failed deployments', () => {
       expect((Deploy.prototype.warn as SinonSpy).firstCall.args[0]).to.equal(expected_error);
     });
 });
+
+describe('deployment secrets', function () {
+  const wildcard_secrets = {
+    'echo': {
+      'a_required_key': 'some_value',
+      'api_port': 3000,
+      'one_more_required_param': 'one_more_value'
```

Comment:
Do we still want to keep all the param references in this file?

Reply:
Nah, I'll update them